dxm/src/test/java/io/pcp/parfait/dxm/PcpMetricInfoV2Test.java

```diff
@@ -1,34 +1,104 @@
 package io.pcp.parfait.dxm;
 
 import io.pcp.parfait.dxm.PcpMetricInfoV2.MetricInfoStoreV2;
+import io.pcp.parfait.dxm.semantics.Semantics;
+import io.pcp.parfait.dxm.types.MmvMetricType;
+import io.pcp.parfait.dxm.types.TypeHandler;
 import org.junit.Before;
 import org.junit.Test;
 import org.mockito.ArgumentMatchers;
+import org.mockito.InOrder;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
 
 import java.nio.ByteBuffer;
 
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+import static org.mockito.ArgumentMatchers.anyLong;
 import static org.mockito.ArgumentMatchers.eq;
 import static org.mockito.Mockito.mock;
 import static org.mockito.Mockito.when;
 import static org.unitils.reflectionassert.ReflectionAssert.assertReflectionEquals;
 
 public class PcpMetricInfoV2Test {
 
+    private static final int STRING_OFFSET = 10;
+    private static final int METRIC_TYPE_IDENTIFIER = 909;
+    private static final int SHORT_HELP_TEXT_OFFSET = 11;
+    private static final int LONG_HELP_TEXT_OFFSET = 12;
+    private static final int EXPECTED_PCP_METRIC_INFO_SIZE = 48;
+    private static final int SEMANTICS_VALUE = 32;
+    private static final int THIS_PCP_METRIC_INFO_OFFSET = 400;
+
     private PcpMetricInfoV2 pcpMetricInfoV2;
 
+    @Mock
+    private PcpString nameAsPcpString;
+
     @Before
     public void setUp() throws Exception {
-        pcpMetricInfoV2 = new PcpMetricInfoV2("name", 1, mock(PcpString.class));
+        MockitoAnnotations.initMocks(this);
+        pcpMetricInfoV2 = new PcpMetricInfoV2("name", 1, nameAsPcpString);
+    }
+
+    @Test
+    public void writeToMmvWriteToTheByteBuffer() {
+        ByteBuffer byteBuffer = ByteBuffer.allocate(48);
+        TypeHandler<Object> typeHandler = mock(TypeHandler.class);
+        MmvMetricType mmvMetricType = mock(MmvMetricType.class);
+        PcpString shortHelpPcpString = mock(PcpString.class);
+        PcpString longHelpPcpString = mock(PcpString.class);
+        Semantics semantics = mock(Semantics.class);
+
+
+        when(nameAsPcpString.getOffset()).thenReturn(STRING_OFFSET);
+        when(typeHandler.getMetricType()).thenReturn(mmvMetricType);
+        when(mmvMetricType.getIdentifier()).thenReturn(METRIC_TYPE_IDENTIFIER);
+        when(shortHelpPcpString.getOffset()).thenReturn(SHORT_HELP_TEXT_OFFSET);
+        when(longHelpPcpString.getOffset()).thenReturn(LONG_HELP_TEXT_OFFSET);
+        when(semantics.getPcpValue()).thenReturn(SEMANTICS_VALUE);
+
+        pcpMetricInfoV2.setTypeHandler(typeHandler);
+        pcpMetricInfoV2.setHelpText(shortHelpPcpString, longHelpPcpString);
+        pcpMetricInfoV2.setSemantics(semantics);
+
+        pcpMetricInfoV2.writeToMmv(byteBuffer);
+
+        byte[] expectedBytes = {
```

Comment:
I think this is a better way to do it than a bunch of verify's on the ByteBuffer. 

However for the ByteChallenged, I wonder if a bit of javadoc is necessary to explain how this works.  I'm not sure if we could inline inside the byte array each important position.  For the _obvious_ one's that map to the constant values above, it might not be needed, but all the multiple positions, and the opaque 'type' values might be good for the reader to undestand?